test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/CSharpSectionTest/BalancesBraces.stree.txt

```diff
@@ -1,48 +1,52 @@
-                SyntaxKind.Text;[(function];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[foo()];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[{];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[return];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[1;];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[})();];
-            Tag block - Gen<None> - 9 - (56:0,56)
-                Markup span - Gen<Markup> - [</script>] - SpanEditHandler;Accepts:Any - (56:0,56) - Tokens:4
-                    SyntaxKind.OpenAngle;[<];
-                    SyntaxKind.ForwardSlash;[/];
-                    SyntaxKind.Text;[script];
-                    SyntaxKind.CloseAngle;[>];
-            Markup span - Gen<Markup> - [ ] - SpanEditHandler;Accepts:Any - (65:0,65) - Tokens:1
-                SyntaxKind.Whitespace;[ ];
-        MetaCode span - Gen<None> - [}] - SpanEditHandler;Accepts:None - (66:0,66) - Tokens:1
-            SyntaxKind.RightBrace;[}];
-    Markup span - Gen<Markup> - [] - SpanEditHandler;Accepts:Any - (67:0,67) - Tokens:1
-        SyntaxKind.Unknown;[];
+RazorDocument - [0..67)::67 - [@section foo { <script>(function foo() { return 1; })();</script> }]
+    MarkupBlock - [0..67)::67
+        MarkupTextLiteral - [0..0)::0 - [] - Gen<Markup> - SpanEditHandler;Accepts:Any
+            Marker;[];
+        CSharpCodeBlock - [0..67)::67
+            RazorDirective - [0..67)::67 - Directive:{section;RazorBlock;Unrestricted}
+                CSharpTransition - [0..1)::1 - Gen<None> - SpanEditHandler;Accepts:None
+                    Transition;[@];
+                RazorDirectiveBody - [1..67)::66
+                    RazorMetaCode - [1..8)::7 - Gen<None> - SpanEditHandler;Accepts:None
+                        Identifier;[section];
+                    CSharpCodeBlock - [8..67)::59
+                        CSharpStatementLiteral - [8..9)::1 - [ ] - Gen<None> - SpanEditHandler;Accepts:Whitespace
+                            Whitespace;[ ];
+                        CSharpStatementLiteral - [9..12)::3 - [foo] - Gen<DirectiveToken {SectionName;Member;Opt:False}> - DirectiveTokenEditHandler;Accepts:NonWhitespace
+                            Identifier;[foo];
+                        MarkupTextLiteral - [12..13)::1 - [ ] - Gen<None> - SpanEditHandler;Accepts:AllWhitespace
+                            Whitespace;[ ];
+                        RazorMetaCode - [13..14)::1 - Gen<None> - AutoCompleteEditHandler;Accepts:None,AutoComplete:[<null>];AtEnd
+                            LeftBrace;[{];
+                        MarkupBlock - [14..66)::52
+                            MarkupTextLiteral - [14..15)::1 - [ ] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                                Whitespace;[ ];
+                            MarkupTagBlock - [15..23)::8 - [<script>]
```

Comment:
Hmm, we might want to special case script tags as a type of element since we parse them differently. Just a thought, whachu think?

Reply:
Hmm I'm not sure. They are still just tags. In this case, will be represent `data-` attributes as a different element because we parse them separately?

Reply:
Spoke offline. We'll do it if it comes up in the future and makes sense.